### ISSUE TITLE
initialLayout={initialLayout} could slow down the application

### ISSUE THREAD

**Issue by wangbourne** [react-native-community/react-native-tab-view#482]
### Current behaviour
My application comes a performance issue. After trying, I found if I remove `initialLayout={initialLayout}`, my application become much faster. What could be the possible reason for this?

By the way, how can I use `TabViewPagerExperimental`?  It does not come in `@types/react-native-tab-view"`  (I'm using Typescript)

### Code sample

```
return (
      <TabViewAnimated
        style={[styles.container, this.props.style]}
        navigationState={this.state}
        renderScene={this._renderScene}
        renderHeader={this._renderHeader}
        onIndexChange={this._handleIndexChange}
        initialLayout={initialLayout}
      />
    );
```

```
_renderScene({ route }: any) {
   // Optimize large number of routes
    const routeIndex = _findIndex(this.state.routes, (item: any) => item.id === route.id);
    if (Math.abs(this.state.index -  routeIndex) > 2) {
      return null;
    }
    //console.log(routeIndex);

    return this.props.renderScene(route);
  }
```
here, `this.props.renderScene` will be called in parent component like

```
//Avoid unnecessary re-renders
renderScene = (route: TabData) => {

    const { sports, match, itemsWithKey, history, handleRouting } = this.props;

    const matches = [...itemsWithKey[route.id].match];

    return (<EventListView match={match} history={history} items={matches} onClick={this.handleEventListClick}/>);
  }
```

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        |   iOS10,11
| react-native          |  0.51.10
| react-native-tab-view |  0.0.74
| node                  |    v9.4.0
| npm or yarn           |    1.5.1 (Yarn)

**Comment by satya164** [comment]
> I found if I remove initialLayout={initialLayout}, my application become much faster. What could be the possible reason for this?

I don't know. What do you mean by faster? Are you talking about initial render time? In that case, I can see how it could make it a bit slower since all scenes are rendered at once when `initialLayout` is specified.

> how can I use TabViewPagerExperimental? It does not come in @types/react-native-tab-view"

I don't maintain the typings and I don't know who does it. You have to find the person who added them.

**Comment by wangbourne** [comment]
As to your answer to `initialLayout={initialLayout}`, you have explained well and thanks :)

By the way, why each scene has to be called twice after render()? I can see following logs from chrome. (My scene rendering is expensive .)

![image](https://user-images.githubusercontent.com/2090084/37342408-3797ebce-2710-11e8-8fda-ed44226a1f71.png)

my code is

```
private _renderScene({ route }: any) {
    const routeIndex = _findIndex(this.state.routes, (item: any) => item.id === route.id);
    if (Math.abs(this.state.index -  routeIndex) > 2) {
      return null;
    }
    console.log(routeIndex);


    return this.props.renderScene(route);
  }

  public render() {
    const { handleIndexChange} = this.props;

    return (
      <TabViewAnimated
        style={[styles.container, this.props.style]}
        navigationState={this.state}
        renderScene={this._renderScene}
        renderHeader={this._renderHeader}
        onIndexChange={this._handleIndexChange}
        initialLayout={initialLayout}
      />
    );
  }
```

**Comment by wangbourne** [comment]
@satya164 The root cause of wrong selected tab indicator is we have `setTimeout` too small. This happens when rendering page is expensive, which finally could make `_isInitial = true` too early 

```
  _setInitialPage = () => {
    if (this.props.layout.width) {
      this._isInitial = true;
      this._scrollTo(
        this.props.navigationState.index * this.props.layout.width,
        false
      );
    }

    setTimeout(() => {
      this._isInitial = false;
    }, 50);
  };
```

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
